tests/api/v1/collectors/heartbeat.js

```diff
@@ -1017,7 +1018,43 @@ describe('tests/api/v1/collectors/heartbeat.js >', () => {
         .then(done)
         .catch(done);
       });
+    });
+
+    describe('added generator >', () => {
+      before(() => tu.toggleOverride('returnUser', true));
+      after(() => tu.toggleOverride('returnUser', false));
+
+      // setup and create a new generator with the createdBy field
+      it('contains the user token', (done) => {
+        const gtPath = '/v1/generatorTemplates';
+        const sgtCopy = JSON.parse(JSON.stringify(sgt));
+        sgtCopy.name = 'iAmNew';
+        sgtCopy.createdBy = userId;
+        sgtCopy.contextDefinition = contextDefinition;
+        gu.createSGtoSGTMapping(sgtCopy, generator2);
+
+        api.post(gtPath)
+        .set('Authorization', userToken)
+        .send(sgtCopy)
+        .expect(constants.httpStatus.CREATED)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
 
+          expect(res.body.createdBy).to.equal(userId);
+          const timestamp = Date.now();
+          return Promise.resolve()
+          .then(() => postGenerator(generator2, [collector1]))
+          .then(() => sendHeartbeat(collector1))
+          .then((res) => {
+            const reencryptedSG = res.body.generatorsAdded[0];
+            expect(reencryptedSG.token).to.be.an('string');
```

Comment:
to.be.a('string')

Reply:
good catch. Updated

Reply:
If you updated it, doesn't look like it got pushed.